lib/services/deviceService.js

```diff
@@ -153,11 +157,85 @@ function sendRegistrations(unregister, deviceData, callback) {
 function processContextRegistration(deviceData, body, callback) {
     var newDevice = _.clone(deviceData);
 
-    newDevice.registrationId = body.registrationId;
+    if (body) {
+        newDevice.registrationId = body.registrationId;
+    }
 
     callback(null, newDevice);
 }
 
+
+function createInitialEntityHandler(deviceData, newDevice, callback) {
```

Comment:
BTW, I just noticed that functions do not include any kind of JSDoc documentation... I guess the linting is not passing due to this... Anyhow, I leave it to your discretion... :) Better days will come when we're able to properly document the code... :p